### ISSUE TITLE
Custom dataset

### ISSUE THREAD

**Issue by AntiLibrary5** [xcyao00/ResAD#13]
Hi,
Thank you for the interesting work. Reading the paper was nice. Using residual features to focus on the differences between an input image and a set of normal reference images. Constraining normal residual features to a hypersphere (with regularization such that anomalous residuals remain in place while only normal ones are pushd to hypersphere) to reduce variations between classes. And finally using a normalizing flow model to learn the probability distribution of normal constrained residual features which can be used to get anomaly maps. Great work and congrats.

But navigating the codebase not as easy. Especially trying on a new dataset. I was wondering if you have plan to have instructions to try the framework on a new dataset? For now especially regarding the dataset `SETTINGS` things are not so flexible. 
```
SETTINGS = {'visa_to_mvtec': VISA_TO_MVTEC, 'mvtec_to_visa': MVTEC_TO_VISA,
            'mvtec_to_btad': MVTEC_TO_BTAD, 'mvtec_to_mvtec3d': MVTEC_TO_MVTEC3D,
            'mvtec_to_mpdd': MVTEC_TO_MPDD, 'mvtec_to_mvtecloco': MVTEC_TO_MVTECLOCO,
            'mvtec_to_brats': MVTEC_TO_BRATS}
```
what about now trying on a new lets say lungs disease dataset? Espcially as you mention:
`this repository is not exactly the same with the method in the NeurIPS conference paper. We further improved the method and open-source the improved code`. But the current code seems much tuned for the Neurips experiments/results.

So I wanted to ask if you have plans to have instructions to try the framework on a new dataset (or there is some branch/repo elsewhere which is more generic) for training/testing with new datasets before doing a great deal of refactoring myself. 
Thank you:)

**Comment by xcyao00** [comment]
Applying the method to a new dataset is not complicated, just write a torch.Dataset class for the new dataset, organize the data according to the format of MVTec. For loading data, you can refer to the MVTEC dataset to write a Dataset class for a custom dataset. In the SETTINGS, we only define some training and testing dataset combinations. You can modify the SETTINGS according to the training dataset and testing dataset you want to set. (Please note that you also need to modify the corresponding training and testing Dataset class in the training code).
